### ISSUE TITLE
Support external groups

### ISSUE THREAD

**Issue by snakebyte91** [redhat-cop/vault-config-operator#168]
I am not sure if it is possible to configure external groups with the resource `JWTOIDCAuthEngineRole` like described here: [Set up Vault external group and AD group](https://developer.hashicorp.com/vault/tutorials/auth-methods/oidc-auth-azure#set-up-vault-external-group-and-ad-group).

If it is not possible it would be a good enhancement of this operator.

**Comment by erlisb** [comment]
Can you elaborate a bit more your request ?
I guess you are referring to the identity secret engine: https://developer.hashicorp.com/vault/api-docs/secret/identity/group ?

**Comment by snakebyte91** [comment]
Yes, this is correct. 

I need to create a group and a group-alias in vault to use a external group and AD groups.

To achive this I have to perform the following steps:

```bash
AZ_GROUP_ID=1abc0123-abcd-1234-abcd-0a123456ab7c
VAULT_GROUP_ID=$(vault write -field=id -format=table identity/group name="user" type="external" policies="user")
VAULT_OIDC_ACCESSOR_ID=$(vault auth list -format=json | jq -r '."oidc/".accessor')
vault write identity/group-alias name="$AZ_GROUP_ID" mount_accessor="$VAULT_OIDC_ACCESSOR_ID" canonical_id="$VAULT_GROUP_ID"
```

**Comment by raffaelespazzoli** [comment]
the identity secret engine is not currently developed in this operator. it would be a great addition.

**Comment by raffaelespazzoli** [comment]
similar to #187 
fixed by #186
